### ISSUE TITLE
BOP extension don't show

### ISSUE THREAD

**Issue by NazakaSyndrome** [Haoming02/sd-webui-old-photo-restoration#5]
Hi,

I've followed the installation steps, but I don't see the "BOP" tab in my list.

I don't get any error messages. Am I missing something?

**Comment by Haoming02** [comment]
Have you restarted your Webui?

**Comment by algarih** [comment]
same here, installed it as instruction, but not "BOP" tab is not showing

**Comment by Haoming02** [comment]
Have you restarted your Webui?

Any error code?

**Comment by algarih** [comment]
yes restarted twice, i cant find a related issue in the cmd when starting

file structure:
extensions > sd-webui-old-photo-restoration > Face_Detection > shape_predictor_68_face_landmarks.dat
extensions > sd-webui-old-photo-restoration > Face_Enhancement > checkpoints (for face_checkpoints.zip)
extensions > sd-webui-old-photo-restoration > Global > checkpoints (for global_checkpoints.zip)

**Comment by Haoming02** [comment]
What's the version of your webui?

**Comment by algarih** [comment]
version: [v1.7.0]

**Comment by Haoming02** [comment]
Hmm...

I just tried again on a fresh install of `v1.8.0` right now. And I still have zero issues...

Sorry, can't recreate without any error or leads...

**Comment by Haoming02** [comment]
I tried on 2 systems from `v1.6.0` to `v1.8.0` and never had this issue T-T

**Comment by Haoming02** [comment]
Can someone try installing this Extension and see if you can get that new Tab at least?

https://github.com/jexom/sd-webui-depth-lib

**Comment by dlovep** [comment]
Installed the extension, still nothing show up, below is the extension list.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/8544a4b7-a684-43ce-a62e-a86239d0076d)

**Comment by Haoming02** [comment]
Not even the **Depth Lib** tab?

**Comment by dlovep** [comment]
> Not even the **Depth Lib** tab?

Nothing at all, no extra tab.

**Comment by Haoming02** [comment]
~~Thank god~~
That means it's the Webui's problem, not mine.

I'll open an issue in the Automatic1111 repo then.

**Comment by dlovep** [comment]
Here are other error on loading main.py
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/c77ed42c-6aa8-46f0-a0ec-aa558c933a75)

**Comment by dlovep** [comment]
I think that's an update issue, I am reinstalling the latest release SD 1.8, will report back if same error. (for BOP)

**Comment by dlovep** [comment]
Reinstall everything fresh from A1111 1.8, still no BOP tab
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/9c71ef23-d26d-4a76-bc3f-7bd4c3550ebd)

**Comment by dlovep** [comment]
Tried Depth Lib, same no new tab
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/4ac89535-e6ba-4e4c-8edc-f00d1e88e8df)

**Comment by Haoming02** [comment]
No error either?

**Comment by Haoming02** [comment]
Opened an [Issue](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/15114) there

Please follow up there with your infos

**Comment by Haoming02** [comment]
@dlovep Can you try installing the Webui on your main drive (`C:`) ?

**Comment by dlovep** [comment]
> @dlovep Can you try installing the Webui on your main drive (`C:`) ?

Tried on C: same, no BOP tab and no Depth Lab tab as well.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/24e83431-ed8c-43be-9ec9-1f5022f950f7)

**Comment by dlovep** [comment]
I tired A1111 1.6 fresh install, Depth Lab and BOP, only Depth Lab shows up.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/971b5f4b-722e-406d-bd36-2a159c2acd25)

**Comment by Haoming02** [comment]
Huh...

And still no Error in the console?

**Comment by dlovep** [comment]
No error at all...
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/a61a1a35-4283-4e7a-85a2-a9db96e0414e)

**Comment by alexsohn** [comment]
I am having the same issue but I lazily installed it using the extension installer in A1111, also I'm on an older version

**Comment by ToKroCZ** [comment]
I have exactly the same problem (1.8.0), but I've also installed DeOldify and that added a new tab, so it seems that not every extension that adds a new tab is broken.

**Comment by dlovep** [comment]
> I am having the same issue but I lazily installed it using the extension installer in A1111, also I'm on an older version

Can you share what version of A1111 ?

**Comment by sergioB79** [comment]
Hi, I tried a few different ways, and I don't see the "BOP" tab on any of my installations.  Other tabs appear and work. Can it be because SD is not installed on C: ? I Don't have space on C: so i can't try it.

**Comment by Haoming02** [comment]
Hello everyone, please check out the latest commit.

The extension is now rewritten into the **Extras** tab instead.

**Comment by dlovep** [comment]
I reinstall 1.7 vanilla and add extension from scratch also paste those zip file in the folder... Still nothing in Extra, also tested 1.8 same...
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/ee317d7c-4ea3-4ab9-809f-41683b047bbf)
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/817839c2-99e1-405c-8fce-b690e76db585)

**Comment by dlovep** [comment]
Once I installed the other extension in 1.7 for testing sd-webui-cn-in-extras-tab
It shows the error message for BOP as well.

![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/4561e927-5a3a-4590-aa31-5d58f69a2180)

**Comment by dlovep** [comment]
Looks like a dlib issue, I tried a few updates still failed to run dlib, even installed a specific dlib https://github.com/Murtaza-Saeed/dlib, no dice.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/dc45f57e-a4b1-4b67-b3da-b81571b597fa)

**Comment by Haoming02** [comment]
If it's about `dlib`, then refer to #4 

**TL;DR:** 
1. Install VS 2022 C++
2. Install CMake
3. Download a pre-built package for `dlib`

**Comment by dlovep** [comment]
> If it's about `dlib`, then refer to #4
> 
> **TL;DR:**
> 
>     1. Install VS 2022 C++
> 
>     2. Install CMake
> 
>     3. Download a pre-built package for `dlib`

YAY ! finally it works, showing in Extra tab:
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/d4ab1b2d-819e-434a-b449-b5bfc2c9350a)

I am suspecting both CMake and dlib causing the issue, once I update the VS 2019 & 2022 to include the Desktop C++ development, checked the box CMake in VS, an I install CMake again using this link https://cmake.org/download/ 
also mark sure the environment path is pointing to (your drive):\ProgramFile\CMake\bin

re run again using pip install dlib, throw some warnings but still go thru.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/119c8314-e5f3-4129-a625-628a4cd7d141)

**Comment by Haoming02** [comment]
So was it always simply caused by `dlib`...? 

Why did the console not mention the failed installation of `dlib` though.

I'm hating the new `install.py` more and more each day... Why did they hide every single information during the installation...

**Comment by dlovep** [comment]
> So was it always simply caused by `dlib`...?
> 
> Why did the console not mention the failed installation of `dlib` though.
> 
> I'm hating the new `install.py` more and more each day... Why did they hide every single information during the installation...

That's the error message only show up in older version like SD1.7, on SD1.8 nothing shows up in console and therefore no-one know what's missing, I tried a few things even with  --enable-insecure-extension-access on SD1.8 still not a single error show up.

**Comment by dlovep** [comment]
Tested update on SD1.8 as expected not a single error on BOP...
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/6958fb51-f0b1-4fcb-8434-abc005dfb506)

But once fixed those 3 below and it run perfect in Extra tab.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/604d4870-23ea-4655-b93a-76b6bda316f7)

Highly likely the dlib is the culprit and blame SD1.8 for hiding the errors.
![image](https://github.com/Haoming02/sd-webui-old-photo-restoration/assets/15388363/cd668b36-3b90-4abd-8eb0-3d9808ef94b6)

**Comment by alexsohn** [comment]
I can also confirm that installing dlib has fixed this error.

**Comment by Haoming02** [comment]
**TL;DR:**
Refer to https://github.com/Haoming02/sd-webui-old-photo-restoration#extension-not-showing-up
